internal/transport/grpchttp2/framer.go

```diff
@@ -299,24 +245,16 @@ func (f *MetaHeadersFrame) Header() *FrameHeader {
 	return f.hdr
 }
 
-// Free is a no-op for MetaHeadersFrame.
-func (f *MetaHeadersFrame) Free() {}
-
 // Framer encapsulates the functionality to read and write HTTP/2 frames.
 type Framer interface {
 	// ReadFrame returns grpchttp2.Frame. It is the caller's responsibility to
-	// call Frame.Free() once it is done using it. Note that once the mem
-	// package gets merged, this API will change in favor of Buffer.Free().
+	// free the underlying buffer when done using the Frame.
 	ReadFrame() (Frame, error)
 	// WriteData writes an HTTP/2 DATA frame to the stream.
-	// TODO: Once the mem package gets merged, data will change type to
-	// mem.BufferSlice.
-	WriteData(streamID uint32, endStream bool, data ...[]byte) error
-	// WriteData writes an HTTP/2 HEADERS frame to the stream.
-	// TODO: Once the mem package gets merged, headerBlock will change type to
-	// mem.Buffer.
-	WriteHeaders(streamID uint32, endStream, endHeaders bool, headerBlocks []byte) error
-	// WriteData writes an HTTP/2 RST_STREAM frame to the stream.
+	WriteData(streamID uint32, endStream bool, data mem.BufferSlice) error
+	// WriteHeaders writes an HTTP/2 HEADERS frame to the stream.
+	WriteHeaders(streamID uint32, endStream, endHeaders bool, headerBlock []byte) error
```

Comment:
What is the reasoning behind not changing the `headerBlock` to be a buffer from the `mem` package?

Reply:
I think there is no point in making `headerBlock` a `mem.Buffer` because of the way the `hpack.Encoder` works. It basically encodes the information into a buffer that lives in the loopy writer, meaning the buffer is always recycled when new fields are encoded. I don't think we need pooling or multiple refs to that specific buffer, so a byte slice should suffice for the use case. What do you think?

Reply:
Makes sense. Thanks.